instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -290,7 +307,21 @@ def _teardown_request(exc):
             # a way that doesn't run `before_request`, like when it is created
             # with `app.test_request_context`.
             return
+        start = flask.request.environ.get(_ENVIRON_STARTTIME_KEY)
+        duration = max(round((default_timer() - start) * 1000), 0)
```

Comment:
Shouldn't all this logic be in `_start_response`?

Reply:
yes, that is also a possible way. But we are starting span in before_request and ending it in teardown_request so I did the same for metrics too. Also [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py#L184) it says that, it is discouraged, I'm also not sure what to prefer. If doing it in start_response is more acceptable then I can make the changes.

Reply:
done.